[Enhancement]

beats-dh opened the pull request.

beats-dh committed:
slighty getSpectators optimization

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Fix possible crash with "creature:canSeeCreature(creature)" lua function.

https: //github.com/opentibiabr/optimized_forgottenserver/commit/71bbe6195c66b18c74291221b76aa794eb2ead4e
Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Fix some potential luascript functions crashes.

https: //github.com/opentibiabr/optimized_forgottenserver/commit/916374e7ceb85e15c8719a04717bf551608221b8
Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Fix spellbook memory leak

https: //github.com/opentibiabr/optimized_forgottenserver/commit/c7d18c513e7cc6817259acbdb4e41dcbaa436961
Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
update?!

https: //github.com/opentibiabr/optimized_forgottenserver/commit/891ea8b3df2f608c93d675f0104b3a0b49b70f4a

- Fix possible "AmbienteActualBrightness = -1" debug
- Make monsters more responsive without needing to adjust their thinking interval

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Small fixes

https: //github.com/opentibiabr/optimized_forgottenserver/commit/00a5abc5144598807180367c3ec7d23a5ee6a8ed
Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Optimize Player::getMuteConditions()

Optimize Player::getMuteConditions() by switching to std::vector to reduce cache misses and allocations

https: //github.com/opentibiabr/optimized_forgottenserver/commit/26678f040027b0af7bde695b55f6a922e4829608
Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Optimize CombatParams::conditionList

Optimize CombatParams::conditionList by switching to std::vector to reduce cache misses and allocations

https: //github.com/opentibiabr/optimized_forgottenserver/commit/88ae5651ce8500edbe7efa5bdb96e563a0f4883d
Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Optimize Player::invitePartyList

Optimize Player::invitePartyList by switching to std::vector to reduce cache misses and allocations
 master

https: //github.com/opentibiabr/optimized_forgottenserver/commit/642c6de99fe849f922382211feb410163cd1602e
Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Optimize Player::storedConditionList

Optimize Player::storedConditionList by switching to std::vector to reduce cache misses and allocations

https: //github.com/opentibiabr/optimized_forgottenserver/commit/619ce0ff8ecebdaf1d32637bfc07e73fd4ea2b48
Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Serialize light into 1 byte instead of 4

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Optimize Container::getContentDescription

Optimize Container::getContentDescription by switching to std::string instead of std::ostringstream

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Fix showUseHotkeyMessage

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Merge spectator loops and use preallocated stackposvector

https://github.com/opentibiabr/optimized_forgottenserver/commit/d08ae89738c2e234a11baf3eb353f46f1425ab3d

https: //github.com/opentibiabr/optimized_forgottenserver/commit/5a974a5113e956efbcc8c4bfaed69f627178ff87
Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Add blockprojectile tile caching

obs

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Refactoring combats

-Use blockprotile tile caching
-Merge checking tilestate's
-Keep MatrixArea in-house of AreaCombat class

https: //github.com/opentibiabr/optimized_forgottenserver/commit/88548b3e6ccdb62c8abfa1da1525e97e81d27d44
Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Keep monsters and npcs in hash map for faster access

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Remove monster from spawnedmap when they die to avoid longer respawn

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Optimize LuaScriptInterface::luaAddEvent

Contain parameters in vector with single allocation.

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
ConnectBlock baninfo micro-optimization

-Create ConnectBlock in-place instead of copying

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Change unnecessary vectors to arrays

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Fix teleport check in map.cpp

Fix teleport check to not issue teleport when changing floor - clients have special packets to handle floor changing so why we don't use them(we can reduce some network usage using them)?

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Some fixes

- Fix displaying hidden monsters on battle list(11.20+)

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Modal Windows optimization

Avoid unnecessary copies

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Fix possible debug

Fix possible debug when changing floors underground

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Optimize towns

Remove double indirection from towns to achieve better caching.

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Optimize houses

Remove double indirection from houses to achieve better caching.

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Vocation cache optimization

-Change cache complexity from O(log n) to O(1)
-Use faster pow function

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Fix Items::getPluralName

Fix Items::getPluralName to avoid possible double "s" at end of string.

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Prevent skill try calculation from underflowing if a player has the b…

…ase skill level

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

Fix underflow issue

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Update tools.h

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Improve Map::placeCreature cache

Remove unnecessary heap allocation

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Improve Monster::getDanceStep cache

obs linha 1398

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
StringVector explodeString optimization

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Fix skill rounding error

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Optimizations and Fixes.

Use std::move to move std::string objects when possible instead of doing copies

beats-dh committed:
Fix target spells and usable items behavior

-Fix target spells by correcting line of sight check
-Fix usable items by correcting line of sight check
-Allow throwing items between multiple floors downwards

beats-dh committed:
Fix monsters behavior

-Monsters remain idle when there appears player they can't see
-Monsters keep their attacks cooldown when they lost target

beats-dh committed:
Fix spells behavior

-Allow wave/beam spells to be called when standing in front of wall
-Fix generated area of effect for wave/beam spells

beats-dh committed:
Fix throwing non-moveable items

beats-dh committed:
Fix pagination when adding new items

-index can't be -1 because when new item arrive to inbox the index already moved upwards
-add if statement incase getItemByIndex failed

beats-dh committed:
Fix Game::playerMoveUpContainer

Don't allow to open more than one instance of container

beats-dh committed:
Fix Player::getWeaponSkill

Ammunition should report SKILL_DISTANCE skill level

beats-dh committed:
Fix ProtocolGame::sendModalWindow

Fix sendModalWindow for QT clients that have swapped default button bytes

beats-dh committed:
update offlineTrainingWindow

beats-dh committed:
Move SpectatorVector reserve

Move SpectatorVector reserve to its constructor

beats-dh committed:
Fix and Optimize CreatureSay

-Fix yelling - players underground should hear only yelling from players that are on the same floor level
-Merge send loops with event loops since it doesn't contain any data race
-Move received spectators vector in Game::internalCreatureSay instead of copying it
-Use getSpectatorsInternal instead of getSpectators since it won't get any cache as it is using non-standard viewport values

beats-dh committed:
Change experience formula

-Cipbia formula have less multiplication which should make it a little faster
-Because it have less multiplication it also don't overflow as often, with Cipbia formula it is possible to reach 1034406 level instead of 717217 with Open Tibia formula

beats-dh committed:
Remove unnecessary boost::lexical_cast

beats-dh committed:
Fix LuaScriptInterface::getField

Enable using LuaScriptInterface::getField for boolean variables

beats-dh committed:
pontecial erro

beats-dh committed:
getSpectators optimization

Map::getSpectators optimization by order of magnitude up to even 10x

https: //github.com/opentibiabr/optimized_forgottenserver/commit/0f1cd7356c318409ae4b2e8495a1d66deea16201
Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
forward_list >> vector

https: //github.com/opentibiabr/optimized_forgottenserver/commit/63f8930449b6f0c0c5d4ddbc21f9d4d2436a7f17
Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Fix Waitlist

https: //github.com/opentibiabr/optimized_forgottenserver/commit/2224d9df1ecc76b131449820012fc77175b01867
Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Fix GlobalEvent revscripts unsafe code

- Fixes possible GlobalEvent null dereference
- Fixes possible GlobalEvent memory leaks

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Fix CreatureEvent revscripts unsafe code

- Fixes possible CreatureEvent null dereference
- Fixes possible CreatureEvent memory leaks

Co-Authored-By: Jakub <38609240+SaiyansKing@users.noreply.github.com>

beats-dh committed:
Optimizations

beats-dh committed:
Replace std::forward_list with std::vector

guispiller reviewed (changes requested):
.

guispiller commented on src/creatures/creature.cpp:
```diff
@@ -448,7 +464,7 @@ void Creature::checkSummonMove(const Position& newPos, bool teleportSummon) cons
 {
 	if (hasSummons()) {
 		// Check if any of our summons is out of range (+/- 2 floors or 30 tiles away)
```
I think there's something missing on this check, because after you summon 2 creatures, and go out of range, they never disappear and you're not able to summon any other creatures unless they got killed. (Main issue too)

beats-dh committed:
Fixed pathfinding magic fields

beats-dh committed:
clear house not exist db

guispiller reviewed (changes requested):
.

beats-dh committed:
sonar

beats-dh committed:
some fixs

dudantas commented on src/creatures/creature.cpp:
```diff
@@ -1292,6 +1341,10 @@ void Creature::removeCondition(ConditionType_t type)
 
 void Creature::removeCondition(ConditionType_t conditionType, ConditionId_t conditionId, bool force/* = false*/)
 {
+	if (!conditionType) {
+		return;
+	}
+
```
```suggestion
```

dudantas commented on src/creatures/players/management/waitlist.cpp:
```diff
@@ -102,55 +103,45 @@ std::size_t WaitingList::getTime(std::size_t slot)
-		if ((g_game().getPlayersOnline() + slot) <= maxPlayers) {
+	auto result = info->findClient(player);
+	if (std::get<1>(result) != std::get<0>(result).end()) {
+		currentSlot = std::get<2>(result);
```
Need test:
```suggestion
	auto [waitList, iterator, size]= info->findClient(player);
	if (iterator != waitList.end()) {
		currentSlot = size;
```

dudantas commented on src/creatures/players/management/waitlist.cpp:
```diff
@@ -102,55 +103,45 @@ std::size_t WaitingList::getTime(std::size_t slot)
+		if ((g_game().getPlayersOnline() + currentSlot) <= maxPlayers) {
 			//should be able to login now
-			info->waitList.erase(it);
+			std::get<0>(result).erase(std::get<1>(result));
```
```suggestion
			waitList.erase(iterator);
```

dudantas commented on src/creatures/players/management/waitlist.cpp:
```diff
@@ -102,55 +103,45 @@ std::size_t WaitingList::getTime(std::size_t slot)
 
 		//let them wait a bit longer
-		it->timeout = OTSYS_TIME() + (getTimeout(slot) * 1000);
+		std::get<1>(result)->timeout = OTSYS_TIME() + (getTimeout(currentSlot) * 1000);
```
```suggestion
		iterator->timeout = OTSYS_TIME() + (getTimeout(currentSlot) * 1000);
```

dudantas commented on src/items/tile.h:
```diff
@@ -57,7 +105,7 @@ class TileItemVector : private ItemVector
 		using ItemVector::reverse_iterator;
 		using ItemVector::const_reverse_iterator;
-    using ItemVector::empty;
+    	using ItemVector::empty;
```
```suggestion
		using ItemVector::empty;
```

dudantas commented on src/lua/functions/creatures/creature_functions.cpp:
```diff
@@ -140,7 +140,11 @@ int CreatureFunctions::luaCreatureCanSeeCreature(lua_State* L) {
+			pushBoolean(L, creature->canSeeCreature(otherCreature));
+		} else {
+			pushBoolean(L, false);
+		}
```
```suggestion
		otherCreature ? pushBoolean(L, creature->canSeeCreature(otherCreature)) : pushBoolean(L, false);
```

dudantas commented on src/lua/functions/creatures/monster/monster_functions.cpp:
```diff
@@ -171,7 +171,11 @@ int MonsterFunctions::luaMonsterIsTarget(lua_State* L) {
+			pushBoolean(L, monster->isTarget(creature));
+		} else {
+			pushBoolean(L, false);
+		}
```
```suggestion
		creature ? pushBoolean(L, monster->isTarget(creature)) : pushBoolean(L, false);
```

dudantas commented on src/lua/functions/creatures/monster/monster_functions.cpp:
```diff
@@ -183,7 +187,11 @@ int MonsterFunctions::luaMonsterIsOpponent(lua_State* L) {
+			pushBoolean(L, monster->isOpponent(creature));
+		} else {
+			pushBoolean(L, false);
+		}
```
Same

dudantas commented on src/lua/functions/creatures/monster/monster_functions.cpp:
```diff
@@ -195,7 +203,11 @@ int MonsterFunctions::luaMonsterIsFriend(lua_State* L) {
+			pushBoolean(L, monster->isFriend(creature));
+		} else {
+			pushBoolean(L, false);
+		}
```
Same

dudantas commented on src/lua/functions/creatures/monster/monster_functions.cpp:
```diff
@@ -207,8 +219,12 @@ int MonsterFunctions::luaMonsterAddFriend(lua_State* L) {
+			pushBoolean(L, true);
+		} else {
+			pushBoolean(L, false);
+		}
```
```suggestion
		if (!creature) {
			pushBoolean(L, false);
			return 1;
		}

		monster->addFriend(creature);
		pushBoolean(L, true);
```

dudantas commented on src/lua/functions/creatures/monster/monster_functions.cpp:
```diff
@@ -220,8 +236,12 @@ int MonsterFunctions::luaMonsterRemoveFriend(lua_State* L) {
+			pushBoolean(L, true);
+		} else {
+			pushBoolean(L, false);
+		}
```
Same

dudantas commented on src/lua/functions/creatures/monster/monster_functions.cpp:
```diff
@@ -268,9 +288,13 @@ int MonsterFunctions::luaMonsterAddTarget(lua_State* L) {
+		pushBoolean(L, true);
+	} else {
+		pushBoolean(L, false);
+	}
```
Same

dudantas commented on src/lua/functions/creatures/monster/monster_functions.cpp:
```diff
@@ -282,8 +306,13 @@ int MonsterFunctions::luaMonsterRemoveTarget(lua_State* L) {
+		pushBoolean(L, true);
+	} else {
+		pushBoolean(L, false);
+	}
```
Same

dudantas commented on src/lua/functions/creatures/monster/monster_functions.cpp:
```diff
@@ -336,7 +365,11 @@ int MonsterFunctions::luaMonsterSelectTarget(lua_State* L) {
+			pushBoolean(L, monster->selectTarget(creature));
+		} else {
+			pushBoolean(L, false);
+		}
```
Same

beats-dh committed:
suggestion Eduardo Dantas

beats-dh committed:
test autoloot

dudantas committed:
improve: rework of Item Attributes for better stability/maintability (#827)

This brings in a major overhaul of the Item Attributes component, addressing long-standing issues in the TFS and OpenTibiaServer project. Adopting class inheritance and harnessing the latest capabilities of C++, this update resolves specific crashes and streamlines future changes to the code. The Attributes class has been separated into its own file, making the code more readable and easier to understand, while also removing any confusing and redundant functions from the Item class. This results in a more stable, maintainable, and user-friendly solution for Item Attributes.

Simplified code by moving to shared_ptr for memory management, removing redundant functions, and revamping key classes such as Attributes, ItemAttribute, and CustomAttribute. The deprecated "set/getSpecialAttribute" functions have been replaced with more efficient "set/getCustomAttribute." This simplified approach leads to better code maintenance and increased stability, for example, Hirelings now utilize the CustomAttribute.

dudantas committed:
fix: client debug and show not targeting player when dying near temple (#841)

The issue arose because the multi-floor was not properly verified, causing incorrect identification of some creatures.

dudantas committed:
fix: some errors and nil value in lua scripts (#830)

murilo09 committed:
feat: added outfit and mount preview feature to the store (#773)

This change adds the new outfit and mount preview feature in the Tibia game store. In addition to the new interface, which offers a better structured and layout, players can now try out all the available outfits and mounts before purchasing them.

ElimarCosta committed:
fix: using a shovel on a stonepile should move to downstairs (#757)

murilo09 committed:
fix: NPC Ongulf (#759)

In the https://github.com/opentibiabr/otservbr-global/pull/781 fixing the `The New Frontier Quest`, mistakenly replaced Ongulf's script with Leeland's script. Reverting to the correct script.

murilo09 committed:
fix: remove money lua function should remove money from bank too (#760)

ElimarCosta committed:
fix: players shouldn't pathfind through fields (#756)

dudantas committed:
fix: change curl version to 1.3 (#847)

ElimarCosta committed:
set target when convince creature (#845)

Convinced creature should attack same target as master
Fixes #539

ElimarCosta committed:
ci: remove action build for drafts (#844)

GitHub Actions will not run on draft prs.

beats-dh committed:
build: deactivate build windows debug (#843)

Glatharth committed:
feat: standardize lever functionality (#648)

This introduces a new standardized function for levers in the game. This function has already been implemented in the Ferumbras' Ascendant Quest, ensuring a consistent and efficient lever experience for players.

beats-dh committed:
fix: high CPU usage issue on linux (#849)

ElimarCosta committed:
fix: house clean-up and item wrap issues (#840)

This fixes two issues related to the '!leavehouse' command. Firstly, all items inside the house should be moved to the owner's mailbox after using the command. Secondly, it should now be possible to wrap unrolled carpets.

Fixes #631: bug causing exposed items after using the '!leavehouse' command, which allowed the new owner of the house to claim those items)."

jacksonie committed:
fix (global datapack): wrath of the emperor teleport (#792)

gpedro committed:
fix: "towns" table database fill issue (#743)

The towns table was not being populated properly after server startup, causing bugs on the web AAC. This aims to fix this issue and ensure proper data filling in the towns database.

beats-dh committed:
fix: rate experience stages (#720)

dudantas committed:
fix: loop in movement "stepin/out" and "addItem/remove" events (#742)

A potential infinite loop was discovered in specific scenarios involving teleport squares or items that teleport players to specific locations (such as when the player logs in). This pull request addresses the issue by disabling the onStepIn event during login and implementing additional security checks to prevent similar bugs from occurring in the future.

beats-dh committed:
improve: added clang format using GHA for code format (#775)

This implements the Clang formatter, a linter, to maintain a consistent code style throughout the code base.

In addition to configuring the Clang formatter, a GH Action has been added to automatically run the linter on the code base every time a push or pull request is made and make a commit to apply the format whenever necessary. This way, we can guarantee that the code is always in the same format.

This also allows us to merge faster, as we can simply apply the linter to any open pull request to resolve any conflicts or changes that may occur during the merge.

carlospess0a committed:
fix: prevent bugs in transfer coins (#714)

Added:
• New method to lua from source;
• Delay in parseTransferCoins.

ElimarCosta committed:
fix: npc "poff" on hit and walk on fields and others (#853)

• Fixed general fixes, NPC "poff" on hit and walk on fields, window projectile blocking and NPC speech bubble.

The NPCs now ignore area damage (not "poffing") and do not traverse condition fields;
Closed windows now obstruct projectiles, while open ones allow them;
The NPC speech bubbles now display correctly with normal icon (...) as default and the trade (small bag) for NPC's with items to trade also with the option to modify the settings in the NPC's Lua configuration file.

dudantas committed:
fix: attribute integer not working (#854)

The "is_integral" was not working, causing integer attributes not to be read.

Complement of commit: https://github.com/opentibiabr/canary/commit/46b2c5ee9e1a6b94a93571e900838025ec50ec70

ElimarCosta committed:
fix: empty reward bags should disappear and others things (#846)

This change aims to resolve the issue with reward bags in the game. Currently, reward bags do not disappear after closing and reopening the reward chest. On the global tibia, it is possible to have more than 100 reward bags, but this leads to strange behaviour if you re-login. Additionally, when multiple bosses are killed at the same time, the rewards are mixed and only one reward bag is created with the combined loot.

The goal is to make the reward bags disappear properly after closing and reopening the reward chest, fix the limit of 100 reward bag, and prevent the mixing of rewards when multiple bosses are killed at the same time.

dudantas committed:
fix: add migrations to create "towns" table (#859)

Complement of: https://github.com/opentibiabr/canary/commit/abdf22f003d875616ecaa34f3370f37f75ba085c

dudantas committed:
feat: implement bosstiary system (#700)

Implements the recently launched Bosstiary system into the game. Bosstiary is an addition to the in-game encyclopedia, allowing players to keep track of which bosses they have defeated and how many times. As players progress through the system, they will receive useful and sophisticated rewards such as equipment loot bonuses, temporary and permanent titles, and the ability to display the defeated boss on a Vigor Podium in their house or guild. Additionally, Bosstiary also features a daily strengthened boss, randomly selected each server save. This change implements all the aforementioned aspects of Bosstiary into the game engine.

dudantas committed:
fix: player auto id (resolves 65536 player GUID max size) (#763)

When the number of registered players in the database exceeds the limit of uint32_t, an overflow occurs and players are unable to move.

Peixonauta2 committed:
fix: gamestore purchase history on ARM architecture (#812)

This commit addresses an issue where the gamestore purchase history was not being properly stored on servers running on ARM architecture. While the problem did not occur on Windows, it affected other systems and was causing frustration for players. This fix ensures that the purchase history is stored correctly on all platforms, without causing any problems for Windows servers.

dudantas committed:
fix: remove shared_ptr from reward and others fixes (#860)

It is a bit more complex to switch to automatic memory management, it is necessary to modify the Game::cleanup function and other functions, as all items are checked there for cleanliness. In the future, maybe we'll modify this, for now it's better to keep it as manual pointer management.

Also fixed some behaviours and issues related to the reward chest.

Complement of: https://github.com/opentibiabr/canary/commit/8a1046a8309848c7f70d505c1401b94f7d180207

Co-author: @ElimarCosta

ElimarCosta committed:
fix: monsters/summons can walk no-walkable items after set unique id (#856)

This fixes an issue where monsters and summons could walk on non-walkable items after their unique ID had been set. The fix ensures that the pathfinding algorithm takes these non-walkable items into account when moving the monsters and summons, preventing them from moving through areas that are designated as non-walkable.

carlospess0a committed:
feat: new lua event "onInventoryUpdate" (#751)

Added a new c++ callback to Lua event on user to handle the inventory item update.

dudantas committed:
fix: depot items not saving (#868)

In the pull request https://github.com/opentibiabr/canary/pull/846 there were many conflicts due to the merge of the clang (#775) pull request, in the resolution of the conflicts, the depot load code ended up being removed, and it went unnoticed, this reverts those changes.

Commit:
improve: handling of modal window helper (#794)

This adds the possibility to create modal window in a more beautiful way.

The order of execution of the calls is: 
• choice
• button
• default

If a choice has a callback it will be activated when you press any button, however the button function will not be activated even if it has one. If another choice does not have a callback, the button callback is called, finally the default callback will be called if there is no callback associated with the button or the choice.

Credits to authors are given in the pull request description.

dudantas committed:
feat: add example script for modal window (#855)

This depends on the pull request: #794

beats-dh committed:
improve: canary datapack functionality with bug fixes and adjust (#665)

Fixes several problems in the canary datapack and tweaked some functions and libs for better behaviour.

LeoTKBR committed:
fix: dangerous depth entrance (change from aid to position) (#863)

murilo09 committed:
improve: fixes cyclopedia and market items description (#826)

Fixing cyclopedia and market items description.

ElimarCosta committed:
fix: interaction with fields and removal of unused flag (#865)

Interaction with fields, fixes #67 
Removal of unused flag changed in #823

dudantas committed:
fix: delete unnecessary script check_inbox.lua (#866)

Probably an inheritance from the OTX, in addition to not being necessary, it calls a table that does not exist "player".
It must have been created at the time that tibia only allowed 10k items in the inbox.

beats-dh committed:
fix: duplicated active events warning and log (#768)

Previously, having two active events for the same date would cause a warning on server load. This pull request fixes the issue and also adds a log for all active events on server load. However, be careful when defining two events with values other than 100, as it will double the final value.

dudantas committed:
fix: canary datapack loading (#873)

The "dofile" was in the wrong file order, fixed this.

dudantas committed:
fix: item not decaying duration and related bugs (#869)

• Fixed item decay with duration (both equip/deEquip and onUse),
• Fixed the look of items with duration, it did not show the duration of some items (example rings, when equipped),
• Functions created in the Item class to avoid code duplication and facilitate the maintenance readability of some systems
• Removed old code from imbuement, which is no longer used
• Removed the decay of items created by loading the ".otbm" map from decay std::map, thus decreasing by about three thousand five hundred items in the decay check.

jacksonie committed:
fix: children of the revolution teleport (#874)

dudantas committed:
fix: custom trainer exit teleport and other things (#872)

FeTads committed:
fix: remove old XML npcs parameter (#835)

Some parameters left over from the old NPCs system in XML, removed and now the NPCs will work.

github-actions[bot] committed:
Code format - (Clang-format)
